src/q8/sigproc-ELEC2900/exam/2019/Juin/All/sigproc-ELEC2900-exam-2019-Juin-All.tex

```diff
@@ -0,0 +1,107 @@
+    $$H(s)=\frac{1}{(s-\e^{\frac{j3\pi}{4}})(s-\e^{\frac{j5\pi}{4}})}$$
+    Give the expression of the corresponding high-pass filter\footnote{The spectral transformation of a filter from low-pass to high-pass is given by $F(\hat{s}) = \frac{\Omega_p\hat{\Omega}_p}{\hat{s}}$.}. Give the corresponding value of $\hat{\Omega}_p$ to implement the filter above.
+    \item What is the cutoff frequency of the digital filter?
+\end{enumerate}
+
+\subsection{LJ2}
+
+\begin{enumerate}
+    \item Remind what are the conditions for a system function $H(z)=\dfrac{\sum_{k=0}^Ma_kz^{-k}}{\sum_{k=0}^Nb_kz^{-k}}$ to be minimum phase.
+    
+    \item Prove mathematically that any rational system function can be written as the product of a minimum phase filter and an all-pass filter.
+    
+    \item As a reminder, the phase lag of a filter is defined as:
+    \begin{equation*}
+        P_L[H](\omega) = - \angle\{H(\e^{j\Omega}) \}
+    \end{equation*}
+    where $\angle \{\e^{j\Omega}\}=\Omega$.
+    
+    Knowing that an all-pass filter has a phase lag which is always positive on the range $[0,\pi]$, prove that among the filters which can produce $|H(\e^{j\Omega})|$, the minimum phase filter is the one with the lowest phase lag on the range $[0,\pi]$.
+\end{enumerate}
+
+\subsection{LJ3}
+
+We are interested in studying the features of the Haar wavelet transform.
+
+\begin{enumerate}
+    \item As a reminder, the \emph{scaling} function is of amplitude 1 between 0 and 1, and the "mother" \emph{wavelet} function is of amplitude 1 between 0 and $\frac{1}{2}$ and -1 between $\frac{1}{2}$ and 1. Deduce what are the impulse responses of the filters $h[n]$ and $g[n]$ used to apply the fast wavelet transform.
+
+    \item Give the schematic which serves to compute the approximation and detail coefficients of the Fast Wavelet Transform for a 1-dimensional signal of $N=2^J$ coefficients. Suppose we can assume the \emph{wavelet crime} and explain this concept.
+    
+    \item What is the computational complexity of this transform? Explain briefly.
+    
+    \item Suppose we have a signal $x[n] \in \mathbb{R}^N$ such that  $x[n] = (x[0],x[1],\dots,x[N-1])$, we build the coefficients $(c[0],c[1],\dots,c[N-1])$, where $c[0]$ is the approximation coefficient at level 0, $c[1]$ is the detail coefficient at level 0, and $(c[2j],\dots,c[2j+1])$ are the detail coefficients at level $j \in (0,J-1)$.
+    
+    For the following signals, what are the indexes where the coefficients $c$ are non-zero?
+    \begin{enumerate}
+        \item $x[n] = (-1)^n$
+        \item $x[n] = 1,\quad\forall n$
+    \end{enumerate}
+    Explain your answers.
+\end{enumerate}
+
+\nosolution
+
+\section{Python}
+
+This part aimed only to give a bonus going from 0 to 1.
+
+\begin{enumerate}
+    \item During the exercise session on image inpainting, what were the names of the algorithms used?
+    \item The following code contains a few errors, list them:
+    \begin{lstlisting}
```

Comment:
```suggestion
    \begin{lstlisting}[language=Python]
```

I'd go even further and suggest you use the `minted` package for better syntax highlighting, but then you need shell-escape and it will generate a couple of auxiliary files so it's up to you.

Reply:
Let's keep it simple for now.